python/pyarrow/table.pxi

```diff
@@ -3549,6 +3549,29 @@ cdef class RecordBatch(_Tabular):
                                                                              row_major, pool))
         return pyarrow_wrap_tensor(c_tensor)
 
+    def copy_to(self, MemoryManager memory_manager):
+        """
+        Copy the entire RecordBatch to destination MemoryManager.
+
+        This copies each column of the record batch to create
+        a new record batch where all underlying buffers for the columns have
+        been copied to the destination MemoryManager.
+
+        Parameters
+        ----------
+        memory_manager : pyarrow.MemoryManager
+
+        Returns
+        ------
+        RecordBatch
+        """
+        cdef:
+            shared_ptr[CRecordBatch] c_batch
+
+        with nogil:
+            c_batch = GetResultValue(self.batch.CopyTo(memory_manager.unwrap()))
```

Comment:
Can `memory_manager` be None here?

Reply:
Ah, yes, with the current design it is a required keyword and so I should protect it from not being None (add a `not None` in the signature). 

But we might want to allow it to be None, and in that case use the source's memory manager as destination device (i.e. copy to the same device)?

Reply:
I don't think copying to the same memory manager is really useful, so we needn't allow None IMHO.